stl/inc/shared_mutex

```diff
@@ -313,35 +422,51 @@ public:
         return _Owns;
     }
 
+    _Pre_satisfies_(_Owns == true)
+    _Pre_satisfies_(_Pmtx != nullptr)
+    _When_(_Owns == false, _Analysis_noreturn_)
+    _When_(_Pmtx == nullptr, _Analysis_noreturn_)
+    _When_(_Owns == true, _Releases_shared_lock_(*this->_Pmtx))
+    _When_(_Owns == true, _Releases_nonreentrant_lock_(*this->_Pmtx))
     void unlock() { // try to unlock the mutex
         if (!_Pmtx || !_Owns) {
             _Throw_system_error(errc::operation_not_permitted);
         }
 
+        _Analysis_assume_lock_held_(_Pmtx);
         _Pmtx->unlock_shared();
         _Owns = false;
     }
 
+    _Swaps_locks_(*this->_Pmtx, *_Right._Pmtx)
+    _Post_satisfies_(_Owns == _Right._Owns)
+    _Post_satisfies_(_Right._Owns == _Owns)
     void swap(shared_lock& _Right) noexcept {
         _STD swap(_Pmtx, _Right._Pmtx);
         _STD swap(_Owns, _Right._Owns);
     }
 
+    _Detaches_lock_(return, *this->_Pmtx)
+    _Post_satisfies_(_Pmtx == nullptr)
+    _Post_satisfies_(_Owns == false)
     mutex_type* release() noexcept {
         _Mutex* _Res = _Pmtx;
         _Pmtx        = nullptr;
         _Owns        = false;
         return _Res;
     }
 
+    _Post_satisfies_(_Owns)
     _NODISCARD bool owns_lock() const noexcept {

```

Comment:
`VSO_0938757_attribute_order` failed again, `_NODISCARD` should be before `_Post_satisfies_`

`_NODISCARD` is everywhere... 

Maybe just put it on the first place and let `clang-format` ugly formatting instead of adding `_NODISCARD` to the statement macro list...

Reply:
My bad. Thanks.

Reply:
Fixed.